src/controllers/csi/provisioner/agent.go

```diff
@@ -120,11 +115,18 @@ func (installAgentCfg *installAgentConfig) installAgent(version, tenantUUID stri
 		}
 	}()
 
-	log.Info("downloading OneAgent package", "architecture", arch)
-	err = dtc.GetAgent(dtclient.OsUnix, dtclient.InstallerTypePaaS, dtclient.FlavorMultidistro, arch, version, tmpFile)
+	arch := dtclient.ArchX86
+	flavor := dtclient.FlavorMultidistro
+	if runtime.GOARCH == "arm64" {
+		arch = dtclient.ArchARM
+		flavor = dtclient.FlavorDefault
+	}
```

Comment:
These are constants anyways and therefore suggest to use [(implicit) build constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints).